perf/impl/https/v0.1/main.go

```diff
@@ -0,0 +1,252 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/binary"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+const (
+	BlockSize = 64 << 10
+)
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	// Read the big-endian bytesToSend value
+	var bytesToSend uint64
+	if err := binary.Read(r.Body, binary.BigEndian, &bytesToSend); err != nil {
+		http.Error(w, "failed to read uint64 value", http.StatusBadRequest)
+		return
+	}
+
+	// Read and discard the remaining bytes in the request
+	io.Copy(io.Discard, r.Body)
+
+	// Set content type and length headers
+	w.Header().Set("Content-Type", "application/octet-stream")
+	w.Header().Set("Content-Length", strconv.FormatUint(bytesToSend, 10))
+
+	// Write the status code explicitly
+	w.WriteHeader(http.StatusOK)
+
+	buf := make([]byte, BlockSize)
+
+	for bytesToSend > 0 {
+		toSend := buf
+		if bytesToSend < BlockSize {
+			toSend = buf[:bytesToSend]
+		}
+
+		n, err := w.Write(toSend)
+		if err != nil {
+			http.Error(w, "Failed write", http.StatusInternalServerError)
+			return
+		}
+		bytesToSend -= uint64(n)
+	}
+}
+
+var zeroSlice = make([]byte, BlockSize) // Pre-made zero-filled slice
+
+type customReader struct {
+	downloadBytes uint64
+	uploadBytes   uint64
+	position      uint64
+}
```

Comment:
Can we just copy-paste the perf implementation we have in the v0.27 folder? The `Read` implementation here looks quite complicated.

Reply:
The `impl/go-libp2p/v0.27` implementation leverages the `io.Writer` interface on `network.Stream` interface. I don't know how to have the Go std http library provide an object that implements `io.Writer` interface for an HTTP request. Instead all the methods that I am aware of require passing an `io.Reader` for an outgoing HTTP request, i.e. `client.Post` and `http.NewRequest`.

@marten-seemann are you aware of an alternative?

Reply:
I left a comment below.